src/server/pfs/server/url.go

```diff
@@ -0,0 +1,207 @@
+package server
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/miscutil"
+	"github.com/pachyderm/pachyderm/v2/src/internal/obj"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/internal/uuid"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	defaultURLTaskSize = 1000
+)
+
+func putFileURL(ctx context.Context, taskService task.Service, uw *fileset.UnorderedWriter, dstPath, tag string, src *pfs.AddFile_URLSource) (n int64, retErr error) {
+	url, err := url.Parse(src.URL)
+	if err != nil {
+		return 0, errors.EnsureStack(err)
+	}
+	switch url.Scheme {
+	case "http":
```

Comment:
[nitpick] I know most of this is copied from `api_server.go`, but these cases could also be combined 
```go
case "https", "http": 
```